### ISSUE TITLE
Hard Requirements for Security, Privacy, Internationalisation and Accessibility Considerations

### ISSUE THREAD

**Issue by nrooney** [w3c/w3process#117]
Every spec document should have the following:

* Security Considerations
* Privacy Considerations
* Internationalisation Considerations 
* Accessibility Considerations.

All these items should be included in each spec to inform implementers and other interested parties on how the spec addresses security, privacy, internationalisation and accessibility, or where potential issues lie. It may also encourage further work, to solve some of the issues identified in these sections. 

This also allows extremely fast review: security / privacy / internationalisation / accessibility experts and teams can review docs extremely fast by just checking whether these sections exist, pushing back to spec authors when they do not.

**Comment by dwsinger** [comment]
While it's important to show that these concerns have been considered and addressed, it's critical to tell implementors what they need to watch for, be careful of, and so on, when implementing this spec. "While this spec. necessarily relays some user information, don't try to be nice to the other end-point by including more information about the user; this increases fingerprinting and invades the user's privacy." The spec. should help implementors do the best job for users.

**Comment by chaals** [comment]
I am happy to require these sections. But...

> This also allows extremely fast review: security / privacy / internationalisation / accessibility experts and teams can review docs extremely fast by just checking whether these sections exist

That only works in the broken situation where there has not been a minimal level of thought about the issues, or it hasn't been written down.

In most cases it turns out people have thought about them, and often found nothing because they were not sure where to look, or not found something because they were thinking along the wrong lines. I think it is more a case that we should be able to focus on more thorough review for a higher level of quality if this allows us to get through the simple stuff faster...

**Comment by chaals** [comment]
On a practical level, the Team can enforce this through pubrules without touching the Process document. Is there a reason why it should be in the formal Process?

**Comment by dwsinger** [comment]
alas, checking for existence is not enough "This spec. does not raise any security or privacy considerations" used to be common text in RFCs. I tend to Chaals' view, that this is a pubrules question rather than process...

**Comment by vfournier17** [comment]
_This also allows extremely fast review: security / privacy / internationalisation / accessibility experts and teams can review docs extremely fast by just checking whether these sections exist, pushing back to spec authors when they do not._

I also disagree with the above language.  This would support the practice of including boilerplate language without thinking about it just to include the sections.

**Comment by chaals** [comment]
To follow up, as an accessibility and privacy reviewer generally the most important task is to find what isn't in the * considerations bits that should be. If this is done right, it can help reviewers further down the line, but we should be very clear about the purpose of any mandatory stuff we require in specs.

**Comment by nrooney** [comment]
I need to write something for this...

**Comment by dwsinger** [comment]
only question seems to be whether it's in the process or publication rules

**Comment by wseltzer** [comment]
We have been addressing in chartering (and in charter template)

**Comment by dwsinger** [comment]
Seems like a charter/pubrules/culture rather than Process (formal) question? Close?

**Comment by vfournier17** [comment]
I have the same concern I mentioned previously, which is that we don't want to provide boilerplate language for folks to include without thinking about what it means.  We're seeing such language show up in new charters, with no clear explanation of what the WG would do to address these issues.

**Comment by LJWatson** [comment]
Given that accessibility is a legal requirement in many regions of the world, I'm not sure we can (or should) leave it to charter language and/or pub rules?

**Comment by dwsinger** [comment]
Does the 'structure and contents of a rec' belong in the process (how we conduct ourselves) at all? And as has been pointed out, merely requiring that they exist doesn't get what we need (that those issues have been thought about). That's why I tend towards pubrules/charter/culture.

**Comment by dwsinger** [comment]
I don't see anything in the process ("how we walk forwards") about what's *in* a Recommendation ("what we're carrying forwards") and I am hesitant to introduce such a concept now. Surely pubrules?

**Comment by dwsinger** [comment]
I propose closing this with no action, and leaving it to pubrules.

**Comment by frivoal** [comment]
I think we can either close with no action (and leave it to pubrules+transition calls). Or if we do want to insist, add a new bullet point in section 6.2.2 Advancement on the Recommendation Track, saying something like that for transitions other than Notes and FPWD, the document should provide information about Security, Privacy, Internationalisation and Accessibility.

I'm leading towards no action.

Agenda+ to try and close on this.

**Comment by chaals** [comment]
I still support closing with no action.
